src/sys/resource.rs

```diff
@@ -0,0 +1,192 @@
+//! Configure the process resource limits.
+use cfg_if::cfg_if;
+
+use crate::errno::Errno;
+use crate::Result;
+
+pub use libc::rlim_t;
+cfg_if! {
+    if #[cfg(all(target_os = "linux", target_env = "gnu"))]{
+        use libc::{__rlimit_resource_t, rlimit, RLIM_INFINITY};
+    }else{
+        use libc::{c_int, rlimit, RLIM_INFINITY};
+    }
+}
+
+libc_enum! {
+    /// The Resource enum is platform dependent. Check different platform
+    /// manuals for more details. Some platform links has been provided for
+    /// earier reference (non-exhaustive).
+    ///
+    /// * [Linux](https://man7.org/linux/man-pages/man2/getrlimit.2.html)
+    /// * [freeBSD](https://www.freebsd.org/cgi/man.cgi?query=setrlimit)
```

Comment:
The preferred spelling is "FreeBSD" (note the capital F)

Reply:
Thanks for the comments. Corrected :)